### ISSUE TITLE
ORDER BY on non-aggregated value in ARRAY_AGG

### ISSUE THREAD

**Issue by derekyidd** [databrickslabs/remorph#121]
Given this example:
```
WITH cte AS (
  SELECT
    id, 
    tag,
    SUM(tag:count) AS item_count
  FROM another_table
)
SELECT
id
, ARRAY_AGG(tag) WITHIN GROUP(ORDER BY item_count DESC) AS agg_tags
FROM cte
GROUP BY 1
```
We get the following error

```
-- [PARSE_SYNTAX_ERROR] Syntax error at or near '('.(line 10, pos 30)
-- 
-- == SQL ==
-- WITH cte AS (
--   SELECT
--     id,
--     tag,
--     SUM(tag.count) AS item_count
--   FROM another_table
-- )
-- SELECT
--   id,
--   ARRAY_AGG(tag) WITHIN GROUP (ORDER BY
-- ------------------------------^^^
--     item_count DESC NULLS FIRST) AS agg_tags
-- FROM cte
-- GROUP BY
--   1
```

I believe the issue is that the ARRAY_AGG can't be ORDER BY with anything other than tag? 

Could we get a more accurate warning/error message?

Is it possible to also solve in remorph?

**Comment by sundarshankar89** [comment]
``` within group``` is not supported, in the interim we can transpile and warn. We will create another issue to solve ```within group``` strategically.

**Comment by sundarshankar89** [comment]
closed by #133
